python/src/worker/response.py

```diff
@@ -30,25 +32,47 @@ def _construct_data_for_upload(self):
                 zipfile.write(self.result.data)
             else:
                 info('Encoding and compressing json work result')
-                json.dump(self.result.data, zipfile)
+                ujson.dump(self.result.data, zipfile)
 
 
+
+        gz_body_bytes = None
+        # If size is less than 2 mb, then send it over notification too
```

Comment:
"Send it directly over socket"*

Reply:
I think we should have a design doc discussing this. It greatly changes the current flow of data and I'd like to see if it's worth the increased complexity of the resulting code.